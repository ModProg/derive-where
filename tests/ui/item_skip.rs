use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Debug)]
#[derive_where(skip_inner)]
enum SkipInnerOnEnum<T> {
	A(PhantomData<T>),
}

#[derive_where(Debug)]
#[derive_where(skip_inner)]
#[derive_where(skip_inner)]
struct DuplicateSkipInner<T>(PhantomData<T>);

#[derive_where(Debug)]
#[derive_where(skip_inner())]
struct EmptySkipInner<T>(PhantomData<T>);

#[derive_where(Debug)]
#[derive_where(skip_inner)]
#[derive_where(skip_inner(Debug))]
struct OverridingSkipInner<T>(PhantomData<T>);

#[derive_where(Debug)]
#[derive_where(skip_inner(Debug))]
#[derive_where(skip_inner)]
struct UnderridingSkipInner<T>(PhantomData<T>);

#[derive_where(Clone; T)]
#[derive_where(skip_inner)]
struct NoSupportedTrait<T>(PhantomData<T>);

#[derive_where(Copy; T)]
#[derive_where(skip_inner(Copy))]
struct UnsupportedTrait<T>(PhantomData<T>);

#[derive_where(Debug)]
#[derive_where(skip_inner(Debug, Debug))]
struct DuplicateTraitSame<T>(PhantomData<T>);

#[derive_where(Debug)]
#[derive_where(skip_inner(Debug))]
#[derive_where(skip_inner(Debug))]
struct DuplicateTraitSeparate<T>(PhantomData<T>);

#[derive_where(Clone; T)]
#[derive_where(skip_inner(Debug))]
struct MissingDeriveTrait<T>(PhantomData<T>);

fn main() {}
