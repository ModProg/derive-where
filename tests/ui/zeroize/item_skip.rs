use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Copy; T)]
#[derive_where(skip_inner(Copy))]
struct UnsupportedTrait<T>(PhantomData<T>);

fn main() {}
