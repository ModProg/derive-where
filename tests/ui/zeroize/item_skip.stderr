error: unsupported skip group, expected one of Clone, Debug, EqHashOrd, Hash, Zeroize
 --> tests/ui/zeroize/item_skip.rs:6:27
  |
6 | #[derive_where(skip_inner(Copy))]
  |                           ^^^^
